packages/x-components/src/x-modules/next-queries/components/__tests__/next-queries.spec.ts

```diff
@@ -52,34 +90,45 @@ describe('testing next queries component', () => {
             >{{ suggestionContentScope.suggestion.query }}</span>
           </template>
         </NextQueries>
-      `,
-      components: {
-        NextQueries
-      }
-    };
+      `;
 
-    nextQueriesWrapper = mount(wrapperComponent, {
-      localVue,
-      store
+    const { wrapper, nextQueries, findTestDataById } = renderNextQueries({
+      template
     });
 
-    const eventSpansList = findTestDataById(nextQueriesWrapper, 'query');
-    const iconsList = findTestDataById(nextQueriesWrapper, 'icon');
+    nextQueries[0].isCurated = true;
+    const eventSpansList = findTestDataById('query');
+    const iconsList = findTestDataById('icon');
+    let highlightIconList = findTestDataById('next-query-highlight');
 
     nextQueries.forEach((nextQuery, index) => {
       expect(eventSpansList.at(index).element.innerHTML).toEqual(nextQuery.query);
       expect(eventSpansList.at(index).element.getAttribute('data-index')).toEqual(`${index}`);
+      expect(highlightIconList.at(index).contains('img')).toBe(false);
       expect(iconsList.at(index)).toBeDefined();
     });
+
+    await wrapper.setProps({ highlightCurated: true });
+    highlightIconList = findTestDataById('next-query-highlight');
+
+    nextQueries.forEach((nextQuery, index) => {
+      expect(highlightIconList.at(index).contains('img')).toBe(!!nextQuery.isCurated);
+    });
   });
 
-  it('renders a button & a custom Next Query', () => {
-    const wrapperComponent = {
-      template: `
+  it('renders a button, a highlighting icon & a custom Next Query', async () => {
+    const template = `
         <NextQueries>
           <template #suggestion="suggestionScope">
```

Comment:
```suggestion
 <template #suggestion="{ suggestion, highlightCurated, index }">
```

Also remove the `suggestionScope.` bellow.